### PR TITLE
CIV-10909 request for reconsideration notification

### PR DESCRIPTION
**Pull request by Omaira-Melo-Hmcts**
### JIRA link (if applicable) ###
[CIV-10909 New property for send a notification in the reconsideration process, when the claim is updated](https://tools.hmcts.net/jira/browse/CIV-10909)

### Change description ###

An email notification is to be sent when the judge rules that the original SDO created and reviewed under the 'request for reconsideration' process has been upheld.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Review by ShwetaTandel-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandler.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            return organisation.get().getName();
+        }
+        return caseData.getApplicant2().getPartyName();
+    }
+
+    public Map<String, String> addPropertiesDef2(final CaseData caseData) {
+        return new HashMap<>(Map.of(
+            CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference(),
+            CLAIMANT_V_DEFENDANT, getClaimantVDefendant(caseData),
+            PARTY_NAME, getLegalOrganizationDef2Name(caseData)
```
Comment (bhagyashreesharma90) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandler.java:115]: for respondent 2 why we need legal org name not the party name ?

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldClaimantNotificationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                             .build())
+                .build();
+
+            handler.handle(params);
+
+            verify(notificationService, times(1)).sendMail(
+                "applicantsolicitor@example.com",
+                TEMPLATE_ID,
+                getNotificationDataMap(caseData),
+                "hearing-fee-unpaid-applicant-notification-000DC001"
```
Comment (bhagyashreesharma90) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldClaimantNotificationHandlerTest.java:75]: hearing-fee needs to be replaced with reconsideration hpheld

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            //Email to respondent1
+            assertThat(targetEmail.getAllValues().get(0)).isEqualTo("sole.trader@email.com");
+            assertThat(emailTemplate.getAllValues().get(0)).isEqualTo("template-id");
+            assertThat(notificationDataMap.getAllValues().get(0)).isEqualTo(getNotificationDataMap(caseData));
+            assertThat(reference.getAllValues().get(0)).isEqualTo("hearing-fee-unpaid-applicant-notification-000DC001");
+            //Email to respondent2
+            assertThat(targetEmail.getAllValues().get(1)).isEqualTo("respondentsolicitor2@example.com");
+            assertThat(emailTemplate.getAllValues().get(1)).isEqualTo("template-id");
+            assertThat(notificationDataMap.getAllValues().get(1)).isEqualTo(getNotificationDataMap2(caseData));
+            assertThat(reference.getAllValues().get(1)).isEqualTo("hearing-fee-unpaid-applicant-notification-000DC001");
```
Comment (bhagyashreesharma90) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandlerTest.java:132]: hearing fee needs to be updated with reconsiderationUpheld

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandler.java (hunk 1 above)
Comment (Omaira-Melo-Hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldDefendantNotificationHandler.java]: I changed it to Respondant2 PartyName

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldClaimantNotificationHandler.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public String camundaActivityId(CallbackParams callbackParams) {
+        return TASK_ID;
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return EVENTS;
+    }
+
+    private CallbackResponse notifyClaimReconsiderationUpheld(CallbackParams callbackParams) {
```
Comment (bhagyashreesharma90) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimReconsiderationUpheldClaimantNotificationHandler.java:53]: add check to see if claimantsolicitoremail is null.. if null don't send email.. we don't need getApplicantEmail method which checks and send email to citizen if solicitor email is null.

File: build.gradle
[the remark this answers is not in the document]
Comment (Omaira-Melo-Hmcts) [reply build.gradle:1]: It should be tested first and then we would create the release tag right?

**Review by bhagyashreesharma90 — DISMISSED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Review by bhagyashreesharma90 — DISMISSED** [review]

File: Jenkinsfile_CNP
Comment (bhagyashreesharma90) [inline Jenkinsfile_CNP:1]: Commenting so we don't merge this file accidentally.

**Review by bhagyashreesharma90 — APPROVED** [review]

**Review by ShwetaTandel-hmcts — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Pull request merged**